### ISSUE TITLE
Dish Wireless Sectors Handled Incorrectly; Notes Too Broadly Applied

### ISSUE THREAD

**Issue by TripEricson** [bluelinepc/signalcheck#48]
(Edited for clarity after doing more math.)

Finally got my device activated on Dish Wireless and it appears that SCP does not know how Dish does its sectors.  Edison HS is 06C28A102/06C28A103/06C28A104, while Bush Hill is 06C28A0ED/06C28A0EE/06C28A0EF.  But it appears that when a note is added, it's applying it to every row beginning with 06C28Axxx; thus, it's treating it like T-Mobile.

It looks like Dish uses sector IDs that are multiples of 21.  The math is clearer when you use decimal.  06C28A0ED = 1814601965, and 06C28A102 = 1814601986.  So take that number, subtract 8, then divide by 21, and that must be the site ID.  For Bush Hill, that means (1814601965 - 8) / 21 = 86409617, and for Edison HS, (1814601986 - 8) / 21 = 86409618.  

Looking at some data I have from Richmond when I convinced an unactivated SIM to work on one of my phones once, it doesn't look like there are any intermediate values between those sector IDs, so it's definitely not like divided by 7 or 3.

Let me know if more information is needed, or if that was unclear.

**Comment by TripEricson** [comment]
I was in Pennsylvania this weekend and managed to get n70 on a few sites.  It looks like they're just offset up a bit, so sector 0xED in n71 becomes 0xF3 in n70.  You can see it on the SignalCheck Map near Chambersburg; the site near Norlo Park is logged with both n71 and n70.

**Comment by TripEricson** [comment]
Here's a post on Reddit about CellMapper having the same problem.

https://www.reddit.com/r/cellmapper/comments/17l7b2v/dish_gnb_calculations_do_not_seem_correct/

That poster claims that n66 fits the pattern also.

Separately, if a proper fix can't be done quickly, I'd rather you just break the site note linkage for Dish entirely rather than leaving its current behavior for any length of time.  I drove across the bridge to Maryland this morning and every new sector I spotted came up with a label in Virginia that was not applicable.  I'm worried I'm going to miss something while editing my database when there are so many incorrect labels applied to new rows.  I also can't make notes while out and about for the same reason.

**Comment by mikejeep** [comment]
Will have a beta update out shortly that should only save site notes for the specific Dish NCI you're entering so it doesn't contaminate more data. Each sector will need its own note for the moment. Will be working on implementing a new method to properly implement this scheme -- thank you for sharing the math!

**Comment by TripEricson** [comment]
Thanks!  I got it.  I'll be out and about tomorrow morning hunting more Dish sites and will give it a shot.

**Comment by TripEricson** [comment]
Unfortunately, the new version didn't fix the problem.

![20231105_Dish](https://github.com/bluelinepc/signalcheck/assets/43734293/2a34ce9a-600f-4d5c-b6ec-d21b79e1e8ec)

**Comment by TripEricson** [comment]
Or, wait, I'm rereading your message and wondering if I misunderstood.  Is it still copying notes when a new sector connects, but just that making a note won't overwrite everything?  If so, I didn't test that because I assumed both behaviors were supposed to have been stopped.

**Comment by mikejeep** [comment]
The new version is only supposed to save a note for the specific Dish NCI/gNB you are connected to when you save the note -- no attempt to link/copy it to any other entries. Were those new notes or notes that were created/saved/linked/copied before you installed the latest update?

**Comment by TripEricson** [comment]
Right, so I misunderstood.  I tested and it will only save a note for a single row, which is good.  However, when I connect to a new sector, it still copies a note from another sector matching the pattern.  You only promised the former, not the latter.  :)

**Comment by mikejeep** [comment]
Well I am glad I stuck to my literal promise, however my intent was to prevent any copying at all.. I'll get on that.. no promises yet.. ;)

**Comment by mikejeep** [comment]
Fix for temporary workaround pending in 4.824b.

**Comment by mikejeep** [comment]
Did the notes stop copying with the latest update @TripEricson ?

**Comment by TripEricson** [comment]
Yes!  Didn't get to test until today.  I successfully made some notes without issue and it didn't copy notes to new sectors.  Good temporary fix!

**Comment by mikejeep** [comment]
Thanks for the update. Will work on getting the proper logic implemented.

**Comment by RyanThaDude** [comment]
It's nice to see SCP addressing this issue and @TripEricson also caught it. For the life of me, I couldn't find anywhere where 4-bit sectors were suppose to be the standard. I could only find a single post on Reddit that states it uses the same method as T-Mobile (12-bit sectors), which was the basis of the formula I created. It may or may not be right, but it's much better than what's going on now! Off topic, but now just to get CM to fix it!

**Comment by RyanThaDude** [comment]
For what it's worth Dish logging is working great and no issues. It's nice to see the correct IDs. I know you're skeptical about copying from other cells with the same IDs, but I have yet to find one that doesn't work. So far, so good. Thanks!

**Comment by TripEricson** [comment]
I wanted to bump this one back to the top of your list.  Dish recently enabled n70 SA here in addition to n71 and n66 SA, and so I've been having a lot of sectors appear with no notes as the n70 sectors connect for the first time, even though I know other sectors on those same sites.  I'm manually copying the labels into the empty rows when I export the file, so it's not a huge deal for me even though it means trying to keep track of what I'm connected to in my head while out and about, but for the less sophisticated user, it's probably rather confusing.
